Flickr/Sources/FlickrAppVCs/FavoritesCoordinator.swift

```diff
@@ -34,33 +40,45 @@ class FavoritesCoordinator {
 
 extension FavoritesCoordinator: FavoritesViewControllerDelegate {
     func selectedImageFromFavoritesVC(imageData: Data, imageTitle: String, imageId: String) {
-        let photoViewerVC = PhotoViewerVC(url: nil, imageTitle: imageTitle, imageId: imageId, imageData: imageData)
+        let photoViewerVC = PhotoViewerVC(url: nil, imageTitle: imageTitle, imageId: imageId, imageData: imageData, dbPool: dbPool)
         photoViewerVC.title = ""
         photoViewerVC.favoritesDelegate = self
         rootNavigationController.pushViewController(photoViewerVC, animated: true)
     }
 
     func getFavoriteImagesFromStorage() -> [FavoriteImageData]? {
-        PersistenceManager.retrieveData()
+        var favoriteimagesArray: [FavoriteImageData] = []
+        do {
+            try dbPool.read { db in
+                let array = try Row.fetchCursor(db, sql: "SELECT * FROM favorites")
+                while let item = try array.next() {
+                    favoriteimagesArray.append(FavoriteImageData(imageId: item["id"], imageData: item["image"], imageTitle: item["name"]))
+                }
+            }
+            return favoriteimagesArray
+        } catch {
+            return nil
+        }
```

Comment:
`dbpool` should directly only be accessed in a persistence manager. This ensures the separation of concerns between the Coordinators and Persistence Mangers.